xtable-core/src/test/java/org/apache/xtable/parquet/TestParquetStatsExtractor.java

```diff
@@ -0,0 +1,177 @@
+        w.endColumn();
+        w.endBlock();
+        w.startBlock(4);
+        w.startColumn(c1, 8, codec);
+        w.writeDataPage(7, 4, BytesInput.from(bytes2), stats, BIT_PACKED, BIT_PACKED, PLAIN);
+        w.endColumn();
+        w.endBlock();
+        w.end(new HashMap<String, String>());
+        return new ParquetFileReader(configuration, path, w.getFooter());
+    }
+
+    @Test
+    public void testToInternalDataFile() {
+        File file = null;
+        ParquetFileReader fileReader = null;
+        InternalDataFile internalDataFile = null;
+        Configuration configuration = new Configuration();
+
+        try {
+            file = new File("./", "test.parquet");
+            fileReader = createParquetFile(file);
+            //statsExtractor toInternalDataFile testing
+            internalDataFile = ParquetStatsExtractor.toInternalDataFile(configuration, fileReader.getPath());
+        } catch (IOException e) {
+            System.out.println(e);
+        }
+        // TODO check if the stats are ok to get as follows
+        List<ColumnStat> testColumnStats = new ArrayList<>();
+        for (BlockMetaData blockMetaData : fileReader.getFooter().getBlocks()) {
+            List<ColumnChunkMetaData> columns = blockMetaData.getColumns();
+            for (ColumnChunkMetaData columnMetaData : columns) {
+                testColumnStats.add(ColumnStat.builder()
+                        .field(InternalField.builder()
+                                .name(columnMetaData.getPrimitiveType().getName())
+                                .parentPath(null)
+                                .schema(schemaExtractor.toInternalSchema(columnMetaData.getPrimitiveType(), columnMetaData.getPath().toDotString()))
+                                .build())
+                        .numValues(columnMetaData.getValueCount())
+                        .totalSize(columnMetaData.getTotalSize())
+                        .range(Range.vector(columnMetaData.getStatistics().genericGetMin(), columnMetaData.getStatistics().genericGetMax()))
+                        .build());
+            }
+        }
+
+        InternalDataFile testInternalFile =
+                InternalDataFile.builder()
+                        .physicalPath("file:/C:/Users/slims/Downloads/XTable/incubator-xtable/xtable-core/test.parquet")//TODO hard coded path to file method
+                        .columnStats(testColumnStats)
+                        .fileFormat(FileFormat.APACHE_PARQUET)
+                        .lastModified(file.lastModified())
+                        .fileSizeBytes(file.length())
+                        .recordCount(8)// TODO remove 8 and replace with record count from file
+                        .build();
+
+        Assertions.assertEquals(
+                true,   InternalDataFile.compareFiles(testInternalFile,internalDataFile));
+    }
+
+    @Test
+    public void main() {
+        testToInternalDataFile();
+    }
```

Comment:
You don't need this since `testToInternalDataFile` is already annotated with `@Test`